desc/plotting.py

```diff
@@ -933,282 +1009,327 @@ def plot_comparison(
+    contourf_kwargs = {}
+    contourf_kwargs["norm"] = matplotlib.colors.Normalize()
+    contourf_kwargs["levels"] = kwargs.get(
+        "levels", np.linspace(np.nanmin(data), np.nanmax(data), ncontours)
     )
-    return label
+    contourf_kwargs["cmap"] = kwargs.get("cmap", "jet")
+    contourf_kwargs["extend"] = "both"
+
+    cax_kwargs = {"size": "5%", "pad": 0.05}
+
+    xx = grid_plot.nodes[:, 2].reshape((grid_plot.M, grid_plot.N), order="F").squeeze()
+    yy = grid_plot.nodes[:, 1].reshape((grid_plot.M, grid_plot.N), order="F").squeeze()
 
+    im = ax.contourf(xx, yy, data, **contourf_kwargs)
+
+    cax = divider.append_axes("right", **cax_kwargs)
+    cbar = fig.colorbar(im, cax=cax)
+    cbar.update_ticks()
+
+    ax.set_xlabel(r"$\zeta_{Boozer}$")
+    ax.set_ylabel(r"$\theta_{Boozer}$")
+    ax.set_title(r"$|\mathbf{B}|~(T)$")
+
+    fig.set_tight_layout(True)
+    return fig, ax
 
-def _name_key(name_dict):
-    """Reconstruct name for dictionary key used in Equilibrium compute methods.
+
+def plot_qs_error(
+    eq,
+    log=True,
+    fB=True,
+    fC=True,
+    fT=True,
+    helicity=(1, 0),
+    L=20,
+    rho=None,
+    ax=None,
+    **kwargs,
+):
+    """Plot quasi-symmetry errors f_B, f_C, and f_T as normalized flux functions.
 
     Parameters
     ----------
-    name_dict : dict
-        name dictionary created by format_name method
+    eq : Equilibrium
+        Object from which to plot.
+    log : bool, optional
+        Whether to use a log scale.
+    fB : bool, optional
+        Whether to include the Boozer coordinates QS error.
+    fC : bool, optional
+        Whether to include the flux function QS error.
+    fT : bool, optional
+        Whether to include the triple product QS error.
+    helicity : tuple, int
+        Type of quasi-symmetry (M, N).
+    L : int, optional
+        Number of flux surfaces to evaluate at. Only used if rho=None.
+    rho : ndarray, optional
+        Radial coordinates of the flux surfaces to evaluate at.
+    ax : matplotlib AxesSubplot, optional
+        Axis to plot on.
 
     Returns
     -------
-    name_key : str
+    fig : matplotlib.figure.Figure
+        Figure being plotted to.
+    ax : matplotlib.axes.Axes or ndarray of Axes
+        Axes being plotted to.
 
     """
-    out = name_dict["base"]
-    if name_dict["sups"] != "":
-        out += "^" + name_dict["sups"]
-    if name_dict["subs"] != "":
-        out += "_" + name_dict["subs"]
-    if name_dict["d"] != "":
-        out += "_" + name_dict["d"]
-    return out
+    if rho is None:
+        rho = np.linspace(1, 0, num=L, endpoint=False)
+    fig, ax = _format_ax(ax)
+
+    data = eq.compute("R0")
+    data = eq.compute("|B|", data=data)
+    R0 = data["R0"]
+    B0 = np.mean(data["|B|"] * data["sqrt(g)"]) / np.mean(data["sqrt(g)"])
+
+    data = None
+    f_B = np.array([])
+    f_C = np.array([])
+    f_T = np.array([])
+    for i, r in enumerate(rho):
```

Comment:
do we need to iterate this, or can it be vectorized by computing it on a dense grid in 3d?